### PR TITLE
Task model

### LINKED ISSUE #20: Create Task model
Create constructor for task / event
Create tag

### PR DESCRIPTION
**Pull request by louislai**
Resolves #20 

This should most of the models required so everyone can work on the commands separately

### CODE REVIEW

File: src/main/java/seedu/toluist/model/Task.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    public Task(String description, LocalDateTime startDateTime) {
+        this(description, startDateTime, null);
+    }
+
+    public Task(String description, LocalDateTime startDateTime, LocalDateTime endDateTime) {
+        assert description != null;
+        this.description = description.trim();
+        this.startDateTime = startDateTime;
+        this.endDateTime = endDateTime;
```
Comment (Melvin-Tan) [inline src/main/java/seedu/toluist/model/Task.java:34]: Need validate startDateTime < endDateTime over here?

File: src/main/java/seedu/toluist/model/Task.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        this.allTags.add(tag);
+    }
+
+    public void removeTag(Tag tag) {
+        this.allTags.remove(tag);
+    }
+
+    public void replaceTags(ArrayList<Tag> tags) {
+        this.allTags = tags;
+    }
```
Comment (Melvin-Tan) [inline src/main/java/seedu/toluist/model/Task.java:58]: Should there be something like a replaceTag function here? This I am not very sure so it's fine to not do it.

**Review by Melvin-Tan — APPROVED** [review]
I approve first, but reply to the comments first before merging :)

File: src/main/java/seedu/toluist/model/Task.java (hunk 2 above)
Comment (louislai) [reply src/main/java/seedu/toluist/model/Task.java:58]: what do you mean?

File: src/main/java/seedu/toluist/model/Task.java (hunk 1 above)
Comment (louislai) [reply src/main/java/seedu/toluist/model/Task.java:34]: right

**Pull request merged**